lib/rdkafka/consumer.rb

```diff
@@ -385,6 +385,37 @@ def seek(message)
       end
     end
 
+    # Lookup offset for the given partitions by timestamp.
+    #
+    # @ param list [TopicPartitionList] The TopicPartitionList with timestamps instead of offets
+    #
+    # @raise [RdKafkaError] When the OffsetForTimes lookup fails
+    #
+    # @return [TopicPartitionList]
+    def offsets_for_times(list)
+      closed_consumer_check(__method__)
+
+      if !list.is_a?(TopicPartitionList)
+        raise TypeError.new("list has to be a TopicPartitionList")
+      end
+
+      tpl = list.to_native_tpl
+
+      response = Rdkafka::Bindings.rd_kafka_offsets_for_times(
+        @native_kafka.inner,
+        tpl,
+        -1 # timeout
```

Comment:
not having timeout may create hard to debug issues. I would opt on providing it even if high (60 seconds for example).